index.bs

```diff
@@ -38,13 +38,64 @@ spec: WEBDRIVER; urlPrefix: https://w3c.github.io/webdriver/
 Introduction {#intro}
 =====================
 
+<em>This section is non-normative.</em>
+
 [[WEBDRIVER|WebDriver]] defines a protocol for introspection and
 remote control of user agents. This specification extends WebDriver by
 introducing bidirectional communication. In place of the strict
 command/response format of WebDriver, this permits events to stream
 from the user agent to the controlling software, better matching the
 evented nature of the browser DOM.
 
+Model {#model}
+==============
+
+This section defines the basic concepts of the WebDriver BiDi
+protocol. These terms are distinct from their representation at the
+<a href=#transport>transport</a> layer, and can be thought of as their
+in-memory representation.
+
+Note: These definitions do not form any normative requirements, rather
+they are used elsewhere as part of such requirements.
+
+A <dfn export>command</dfn> is sent by the client to the [=remote
+end=], requesting it to perform some action. A [=command=] consists
+of:
+ - An <dfn export for=command>id</dfn> (integer) to be used in the
+    resulting [=response=] or [=error=].
+ - A <dfn export for=command>domain</dfn> (string)
+ - A <dfn export for=command>name</dfn> (string)
+ - <dfn export for=command>parameters</dfn> ([=map=] with string
+    <a for=map>keys</a> and any type as <a for=map>values</a>)
+
+A <dfn export>response</dfn> is sent by the [=remote end=] to the
+client in response to a specific [=command=], and signifies the
+success of that command. A [=response=] consists of:
+ - An <dfn export for=response>id</dfn> (integer) equal to the
+    corresponding [=command=]'s <a for=command>id</a>.
+ - A <dfn export for=response>result</dfn> (any type)
```

Comment:
When you say "type" here, what are you imagining? I initially thought "JSON type" (string, number, object, etc.), but I'd like to be able to define types in the spec ("RealmId", "ContextId", "JSValue", etc.), which might have complex serilizations. If we mean the latter I think I"m OK with this, but we'll need to define "type" later.

Reply:
I mean any https://infra.spec.whatwg.org/#primitive-data-types or other type that we define. We will indeed need to define exactly how to turn a hopefully-JSON-byte-sequence into these constructs.

Reply:
Yes, I strongly agree with the idea of working as much as possible at the highest level types.